internal/provider/network/routed_resource.go

```diff
@@ -0,0 +1,540 @@
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	govcdtypes "github.com/vmware/go-vcloud-director/v2/types/v56"
+
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/client"
+	"github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/helpers/stringpm"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource                = &networkRoutedResource{}
+	_ resource.ResourceWithConfigure   = &networkRoutedResource{}
+	_ resource.ResourceWithImportState = &networkRoutedResource{}
+)
+
+// NewNetworkRoutedResource is a helper function to simplify the provider implementation.
+func NewNetworkRoutedResource() resource.Resource {
+	return &networkRoutedResource{}
+}
+
+// networkRoutedResource is the resource implementation.
+type networkRoutedResource struct {
+	client *client.CloudAvenue
+}
+
+type networkRoutedResourceModel struct {
+	ID            types.String `tfsdk:"id"`
+	Name          types.String `tfsdk:"name"`
+	Description   types.String `tfsdk:"description"`
+	EdgeGatewayID types.String `tfsdk:"edge_gateway_id"`
+	InterfaceType types.String `tfsdk:"interface_type"`
+	Gateway       types.String `tfsdk:"gateway"`
+	PrefixLength  types.Int64  `tfsdk:"prefix_length"`
+	DNS1          types.String `tfsdk:"dns1"`
+	DNS2          types.String `tfsdk:"dns2"`
+	DNSSuffix     types.String `tfsdk:"dns_suffix"`
+	StaticIPPool  types.Set    `tfsdk:"static_ip_pool"`
+}
+
+type staticIPPool struct {
+	StartAddress types.String `tfsdk:"start_address"`
+	EndAddress   types.String `tfsdk:"end_address"`
+}
+
+var staticIPPoolAttrTypes = map[string]attr.Type{
+	"start_address": types.StringType,
+	"end_address":   types.StringType,
+}
+
+// Metadata returns the resource type name.
+func (r *networkRoutedResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_" + "routed"
+}
+
+// Schema defines the schema for the resource.
+func (r *networkRoutedResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Provides a VMware Cloud Director Org VDC routed Network. This can be used to create, modify, and delete routed VDC networks.",
```

Comment:
```suggestion
		MarkdownDescription: "Provides a CloudAvenue Org VDC routed Network. This can be used to create, modify, and delete routed VDC networks.",
```